uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1017.php

```diff
@@ -46,6 +50,62 @@ private function up1017_migrateLocationTypes() {
     }
   }
 
+  /**
+   * Sets the default location type
+   *
+   * @param string $locationType
+   */
+  private function up1017_setDefaultLocationType($locationType) {
+    $locationTypeId = $this->up1017_getLocationTypeID($locationType);
+
+    civicrm_api3('LocationType', 'create', [
+      'id' => $locationTypeId,
+      'is_default' => TRUE
+    ]);
+  }
+
+  /**
+   * Sets all given location types as reserved
+   *
+   * @param array $locationTypes
+   */
+  private function up1017_reserveLocationTypes($locationTypes) {
+    foreach ($locationTypes as $locationType) {
```

Comment:
A little bit of optimization, would be good to store the location id's in an array and then  make the API call to set them as reserved once. Not sure it would make much difference here though since we are not updating many records.

Reply:
I don't think the API can handle multiple updates (i.e. `'id' => ['IN' => [1,2]]`) and to be honest I avoid chaining whenever possible. I don't like the syntax, readability and idea that you don't have control over what happens at each stage (e.g. handling a failed get request when chained to create request). If you think it is necessary I can change it, but since we're just passing two location types here I'd prefer to leave it as it is.

Reply:
Its fine as is. Forgot the API doesn't allow IN for updates.